packages/server/src/handlers/listAccounts.ts

```diff
@@ -0,0 +1,69 @@
+import { Account, CurrencyType } from "@ledgerhq/wallet-api-core";
+import type { SimpleJSONRPCMethod } from "json-rpc-2.0";
+import type { ServerParams } from "..";
+import globStringToRegex from "../utils/globStringToRegex";
+
+const listAccounts: SimpleJSONRPCMethod<ServerParams> = (
+  params,
+  serverParams
+): Account[] => {
+  if (!serverParams || !serverParams.accounts || !serverParams.currencies) {
+    return [];
+  }
+
+  const { accounts: serverAccounts, currencies: serverCurrencies } =
+    serverParams;
+
+  if (!params) {
+    return serverAccounts;
+  }
+
+  const { params: methodParams } = params as {
+    params?: {
+      /**
+       * Include tokens accounts in the results
+       */
+      includeTokens?: boolean;
+      /**
+       * Select a set of currencies by id to filter accounts agains.
+       * Globing is enabled
+       */
+      currencies?: string[];
+    };
+  };
+
+  if (!methodParams) {
+    return serverAccounts;
+  }
+
+  const { includeTokens, currencies } = methodParams;
+
+  const listedAccounts = includeTokens
+    ? serverAccounts
+    : serverAccounts.filter((account) => {
+        // TODO: we should store `currencies` in a Map to quicky access a currency by it's id
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
+        const currency = serverCurrencies.find(
+          (c) => c.id === account.currency
+        );
+
+        return currency?.type === CurrencyType.CryptoCurrency;
+      });
```

Comment:
Could we have a `filteredServerAccounts`, `serverAccountsWithTokens`
 or `serverAccountsNoTokens` already done and shared with `serverParams` ?

Because I feel like this is going to be the most expansive filter if we have a lot of tokens

Reply:
Updated in d7f1c562c40ddfcfa6ffe999222cd8b9da5dfc8b

PS: By the way, I don't get where the class properties like `currencies`, `accounts` or the new `currenciesNoToken` and `accountsNoToken` used in `ServerParams` are actually linked to the JSON-RPC server instance 🤔 

For the 2 new properties added, I get a warning for each one saying it's `declared but its value is never read`